message-index/messages/GHC-53786/case-expr-in-fun-def/before/CaseExprInFunctionDef.hs

```diff
@@ -0,0 +1,3 @@
+module CaseExprInFunctionDef where
+
+f (case x of { Nothing -> 0; Just a -> a }) = a
```

Comment:
I wonder if this having a type sig makes it less clear, hmmm

but it would be nice to be consistent?

Reply:
Since this is completely nonsensical, a type signature corresponding to the fixed version might make it more confusing.
This error can really only occur if the user lacks understanding of basic syntax, so I think simpler is better.